### PR TITLE
Copy the permissions along with file for jboss module

### PR DESCRIPTION
**Pull request by xcompass**
##### SUMMARY

The deployment file is copied with file content only. The file
permission is set to 440 and belongs to root user. When the
JBossI(Wildfly) server is running under non root account, it can't read
the deployment file.

With is fix, the correct permission can be set from previous task for JBoss
server to pickup the deployment file.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

Jboss

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)? Thanks.

**Comment by xcompass** [comment]
@felixfontein done.

File: plugins/modules/web_infrastructure/jboss.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def main():
             # Clean up old failed deployment
             os.remove(os.path.join(deploy_path, "%s.failed" % deployment))
 
-        shutil.copyfile(src, os.path.join(deploy_path, deployment))
+        shutil.copy(src, os.path.join(deploy_path, deployment))
```
Comment (Ajpantuso) [inline plugins/modules/web_infrastructure/jboss.py:145]:
```suggestion
        module.preserved_copy(src, os.path.join(deploy_path, deployment))
```

Taking this a step further I would use the `Ansible` provided [utilities](https://github.com/ansible/ansible/blob/58b03be4178fda12a7d082ae9fd54104229e9d99/lib/ansible/module_utils/basic.py#L1623) if appropriate to preserve as much of the source file's attributes as possible.

File: changelogs/fragments/3367-copy-permissions-along-with-file-for-jboss-module.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+bugfixes:
+  - jboss - fix the deployment file permission issue when Jboss server is running 
+    under non-root user. The deployment file is copied with file content only. The 
+    file permission is set to 440 and belongs to root user. When the
+    JBossI(Wildfly) server is running under non-root user, it is unable to read
+    the deployment file.
```
Comment (Ajpantuso) [inline changelogs/fragments/3367-copy-permissions-along-with-file-for-jboss-module.yml:6]:
```suggestion
    JBoss I(WildFly) server is running under non-root user, it is unable to read
    the deployment file (https://github.com/ansible-collections/community.general/pull/3426).
```

File: plugins/modules/web_infrastructure/jboss.py [hunk 3]
```diff
@@ -153,7 +153,7 @@ def main():
     if state == 'present' and deployed:
         if module.sha1(src) != module.sha1(os.path.join(deploy_path, deployment)):
             os.remove(os.path.join(deploy_path, "%s.deployed" % deployment))
-            shutil.copyfile(src, os.path.join(deploy_path, deployment))
+            shutil.copy(src, os.path.join(deploy_path, deployment))
```
Comment (Ajpantuso) [inline plugins/modules/web_infrastructure/jboss.py:156]: Same here.

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Just a few comments to start with.

**Comment by xcompass** [comment]
Updated according to suggestion.

File: changelogs/fragments/3426-copy-permissions-along-with-file-for-jboss-module.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+bugfixes:
+  - jboss - fix the deployment file permission issue when Jboss server is running 
+    under non-root user. The deployment file is copied with file content only. The 
+    file permission is set to 440 and belongs to root user. When the
+    JBossI(Wildfly) server is running under non-root user, it is unable to read
+    the deployment file. (https://github.com/ansible-collections/community.general/pull/3426)
```
Comment (Ajpantuso) [inline changelogs/fragments/3426-copy-permissions-along-with-file-for-jboss-module.yml:6]:
```suggestion
    file permission is set to ``440`` and belongs to root user. When the
    JBoss ``WildFly`` server is running under non-root user, it is unable to read
    the deployment file (https://github.com/ansible-collections/community.general/pull/3426).
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Just some more updates to the changelog fragment.

**Review by Ajpantuso — APPROVED** [review]
LGTM, @xcompass just confirming if you've tested in your environment successfully?

**Comment by xcompass** [comment]
@Ajpantuso Yes, I'm using it as local module in my environment.

**Comment by felixfontein** [comment]
@xcompass thanks for fixing this!
@Ajpantuso thanks for reviewing!

**Pull request merged**